### PR TITLE
Optionally read templates from STDIN

### PR DESCRIPTION
**Pull request by nVitius**
This addresses #10

### CODE REVIEW

**Comment by nVitius** [comment]
First time I've contributed to anything written in Rust. Comments are welcome.

**Comment by jimmycuadra** [comment]
Thanks very much for the contribution! Only a couple things I'd change:
1. Although the style you use for importing std::io is common, I prefer to import functions directly, so in this case `use std::io::{Read, stdin}`.
2. Rather than specifying an option to tell ktmpl to read from stdin, it'd be great if it simply tried to read from stdin, and if there is data, treat it as the template, falling back to looking for a file path argument. This would be similar to how tools like grep behave (you can give it a file name or pipe data into it).

**Comment by nVitius** [comment]
Thank you for the feedback! I've made the changes you suggested. Let me know what you think.

File: src/main.rs [hunk 1]
```diff
@@ -75,10 +73,15 @@ fn real_main() -> Result<(), String> {
...
-    let filename = matches.value_of("template").expect("template wasn't provided");
-    let mut file = try!(File::open(filename).map_err(|err| err.description().to_owned()));
     let mut template_data = String::new();
-    try!(file.read_to_string(&mut template_data).map_err(|err| err.description().to_owned()));
+    if matches.is_present("template") {
+      let filename = matches.value_of("template").expect("template wasn't provided");
+      let mut file = try!(File::open(filename).map_err(|err| err.description().to_owned()));
+      try!(file.read_to_string(&mut template_data).map_err(|err| err.description().to_owned()));
+    } else {
+      stdin().read_to_string(&mut template_data).ok().expect("Error when reading from stdin");
```
Comment (jimmycuadra) [inline src/main.rs]: Let's return an `Err` here instead of panicking so the error output is nicer.

Comment (nVitius) [reply src/main.rs]: Okay, I changed it to use `try!`

**Comment by jimmycuadra** [comment]
Thanks so much! I'll get a new release out soon.

**Comment by jimmycuadra** [comment]
I decided to change it again so that using `-` as the filename means to read from stdin, similar to how kubectl works. Thanks again for you work! Now released in [ktmpl 0.5.0](https://github.com/InQuicker/ktmpl/releases/tag/0.5.0).

**Pull request merged**